src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/CompareConverter.shared.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	/// Converts an object that implements IComparable to an object or a boolean based on a comparaison.
+	/// </summary>
+	public class CompareConverter : CompareConverter<object>
+	{
+	}
+
+	/// <summary>
+	/// Converts an object that implements IComparable to an object or a boolean based on a comparaison.
+	/// </summary>
+	public class CompareConverter<TObject> : ValueConverterExtension, IValueConverter
+	{
+		[Flags]
```

Comment:
if we use flags, should we add the shift notation (1 << 0 )notation into the values?

Reply:
What we will gain from using the shift notation? I mean we are not going to do an "and" or an "or operation on flags, are we? the only thing is that we can build GreaterOrEqual  and  SmallerOrEqual  from: ` GreaterOrEqual = Greater OR Equal` and `SmallerOrEqual =Smaller OR Equal`, or i am missing something else ?

Reply:
It's needed, from the [guidelines](https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-5.0#guidelines-for-flagsattribute-and-enum)

Reply:
My bad I didn't got your point at the beginning (I thought combining flags that could be combined), sure I will update and give them specific values.